src/org/jrubyparser/lexer/Lexer.java

```diff
@@ -1606,6 +1610,10 @@ private int ampersand(boolean spaceSeen) throws IOException {
             yaccValue = new Token("&", getPosition());
             setState(LexState.EXPR_BEG);
             return Tokens.tOP_ASGN;
+        case '.':
+            setState(LexState.EXPR_BEG);
+            yaccValue = new Token("&.", getPosition());
+            return Tokens.tLONELY;    
```

Comment:
This is called Tokens.tANDOT in both MRI and JRuby core.  I suggest renaming it to make comparisons simpler.